add babel esm and umd exports

redallen opened the pull request:
Fix https://github.com/openshift/console/pull/2232 upstream.

priley86 commented on package.json:
```diff
@@ -22,9 +22,11 @@
-    "prebuild": "yarn test --ci && shx rm -rf dist",
-    "build": "NODE_ENV=production yarn build:js && yarn build:sass",
-    "build:js": "babel src --config-file ./config/babel.config.js --out-dir dist/js --only 'src/**/*.js' --ignore 'src/jest,src/cosmos,**/tests/**,**/fixtures/**'",
+    "build": "yarn build:babel && yarn build:sass",
```
should `NODE_ENV=production` and `&& shx rm -rf dist` be preserved?

redallen replied:
`NODE_ENV=production` doesn't matter. I added back the `prebuild` step, though.

priley86 commented:
cc: @suomiy @mareklibra - it seems you'll need to export ESM to treeshake Console UI bundle. Thanks @redallen !

redallen commented on package.json:
```diff
@@ -43,7 +46,9 @@
   },
   "dependencies": {
     "@patternfly/react-console": "1.x",
+    "babel-plugin-transform-es2015-modules-umd": "^6.24.1",
```
Done.

redallen commented on config/eslint.browser.js:
```diff
@@ -59,5 +59,6 @@ module.exports = {
         groups: [['builtin', 'external'], ['internal', 'parent', 'sibling', 'index']],
       },
     ],
+    'import/no-cycle': 'off'
```
Done.

redallen commented:
@suomiy I fixed the tests and addressed your feedback. We needed a CJS babel config for Jest to run the tests on Node, so I passed it manually to Jest.

CreateVMWizard.test.js seems to have something weird going on with it:
```
console.error node_modules/react/cjs/react.development.js:225
Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

Check the render method of `WizardPatternBody`.
    in WizardPatternBody (created by WizardPattern)
    in div (created by ModalBody)
    in ModalBody (created by WizardBody)
    in WizardBody (created by WizardPattern)
    in div (created by Wizard)
```

but I think that's outside the scope of this PR and is probably a PatternFly problem.

mareklibra commented on package.json:
```diff
@@ -23,8 +23,11 @@
-    "build:js": "babel src --config-file ./config/babel.config.js --out-dir dist/js --only 'src/**/*.js' --ignore 'src/jest,src/cosmos,**/tests/**,**/fixtures/**'",
+    "build": "yarn build:babel && yarn build:sass",
+    "build:babel": "concurrently \"yarn build:babel:esm && yarn build:babel:umd\" \"yarn build:babel:cjs\"",
+    "build:babel:cjs": "babel --config-file ./config/babel.config.js --source-maps --extensions \".js,.ts,.tsx\" src --out-dir dist/js --presets=@babel/env",
```
Can we remove the test-related code from the resulting `dist`?

suomiy replied:
+1 that could decrease a size a bit

redallen replied:
Fixed. It's rather silly babel didn't respect my `ignore: []` in the `babel.config.js` :(

mareklibra commented on config/eslint.browser.js:
```diff
@@ -59,5 +59,6 @@ module.exports = {
         groups: [['builtin', 'external'], ['internal', 'parent', 'sibling', 'index']],
       },
     ],
+    'import/no-cycle': 'off',
```
We should remove the dependency cycles in a follow-up.
Brief look - this should not be so difficult.

redallen commented:
@mareklibra Added the ignores.

mareklibra commented:
Thank you for the patch!
